### DISCUSSION TITLE
Code structure question

### DISCUSSION THREAD

**Discussion by adriesse** [pvlib/pvlib-python#1438] [General]
I want to add a set of related functions and helper functions to pvlib.irradiance.  What do you all think about using a class with static methods as discussed [here](https://stackoverflow.com/questions/38758668/grouping-functions-by-using-classes-in-python).  There will be an array of constants too, which would be nicely contained inside the class.

**Comment by adriesse** [comment]
Discussion is in Google groups.

**Comment by mikofski** [comment]
I was going to add something, but my thoughts echo what Kevin and Cliff already said over in the Google groups thread.

**Reply by markcampanelli** [reply]
All other things being equal, I’d prefer an irradiance sub-package with separate modules containing constants, related functions, etc.
